ha/cli/exec/supportBundleExecutor.py

```diff
@@ -0,0 +1,148 @@
+        return True
+
+    def execute(self) -> str:
+        self.process_request()
+
+
+    def process_request(self, bundle_action="create"):
+        """
+        Process support Bundle request
+
+        Args:
+            action (string): Action to be taken for support bundle.
+            args (dict): Parameter for support bundle
+        """
+        getattr(self, bundle_action)()
+
+    def create(self):
+        """
+        Create support bundle on given path.
+        """
+        self._add_more_log()
+        export_path = self._export_logs(self._check_files())
+        print(f"Successfully created support bundle at {export_path}")
+        Log.info(f"Successfully created support bundle at {export_path}")
+
+    def _add_more_log(self):
+        """
+        Collect Extra Log
+        """
+        cmd_list = [const.PCS_STATUS, const.PCS_FAILCOUNT_STATUS]
+        with open(const.HA_CMDS_OUTPUT, "w") as file:
+            for cmd in cmd_list:
+                _err = ""
+                _proc = SimpleProcess(cmd)
+                _output, _err, _rc = _proc.run(universal_newlines=True)
+                file.write("\n\n\n********************************\n")
+                file.write(f"Command: {cmd}\nReturn Code: {_rc}\nOutput:\n")
+                file.write(f"{_output}")
+                file.write(f"Error if any: {_err}\n\n\n")
+
+    def _check_files(self, ha_log_list=const.SUPPORT_BUNDLE_LOGS):
+        """
+        Check all support bundle file for HA
+        and provide list of available file
+        """
+        ha_logs = list(ha_log_list)
+        with open(const.SUPPORT_BUNDLE_ERR, "w") as file:
+            for _path in ha_log_list:
+                if not os.path.exists(_path):
+                    ha_logs.remove(_path)
+                    file.write(f"For support bundle {_path} is not available")
+        return ha_logs
+
+    def _export_logs(self, ha_logs):
+        """
+        Create tar for support bundle logs
+
+        Args:
+            ha_logs (list): List of support bundle files
+        """
+        path = os.path.join(self._path, "HA")
+        os.makedirs(path, exist_ok=True)
+        # Generate Tar file for Logs Folder.
+        tar_file_name = os.path.join(path, f"ha_{self._id}.tar.gz")
+        Tar(tar_file_name).dump(ha_logs)
+        if not os.path.exists(tar_file_name):
+            raise SupportBundleError(f"Failed to create support bundle at {tar_file_name}")
+        return tar_file_name
+
+class CreateSupportBundleExecutor(CreateExecutor):
+
+    def __init__(self):
+        """
+        This Class generates the Support Bundle for Component cortx HA V2.
+        This method will generate bundle for HA V2 and include the logs in it.
+        """
+        super(CreateSupportBundleExecutor, self).__init__()
+
+    def _add_more_log(self):
+        """
+        Collect HA commands Log
+        """
+        pass
+
+    def _check_files(self, ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS):
```

Comment:
Modify
```
def _check_files(self, ha_log_list=None):
          if ha_log_list == None:
                 ha_log_list = const.CORTX_SUPPORT_BUNDLE_LOGS
```

Reply:
@ajay-paratmandali what's the trick? Look equal.

Reply:
```
    def _check_files(self, ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS):
```
It provide some codacy issue to use constant variable assign in parameter 
To address codacy issue we can change format 
```
ha_log_list=const.CORTX_SUPPORT_BUNDLE_LOGS
```
Here we can assign string or none but not variable as per codacy rule